medusa/server/api/v2/cache.py

```diff
@@ -0,0 +1,143 @@
+        else:
+            self.api_finish(data={},error="Error, need to provide")
+
+        data = [{'exception_id': row[0],
+                 'indexer': row[1],
+                 'indexer_id': row[2],
+                 'show_name': row[3],
+                 'season': row[4],
+                 'custom': row[5]}
+                for row in exceptions]
+
+        self.api_finish(data=data)
+
+    def put(self, *args, **kwargs):
+        """Update show information.
+
+        :param show_id:
+        :type show_id: str
+        """
+        try:
+            exception_id = int(kwargs.pop('row_id', 'throwme'))
+        except (ValueError, TypeError):
+            return self.api_finish(status=400, error="Require a valid exception row id.")
+
+        data = json_decode(self.request.body)
+
+        if not all([data.get('indexer_id'),
+                    data.get('season'),
+                    data.get('show_name'),
+                    data.get('indexer')]):
+            return self.api_finish(status=400, error="Invalid post body, can't update")
+
+        cache_db_con = db.DBConnection('cache.db')
+        last_changes = cache_db_con.connection.total_changes
+        cache_db_con.action(b'UPDATE scene_exceptions'
+                            b' set indexer = ?'
+                            b', indexer_id = ?'
+                            b', show_name = ?'
+                            b', season = ?'
+                            b', custom = 1'
+                            b' WHERE exception_id = ?',
+                            [data.get('indexer'),
+                             data.get('indexer_id'),
+                             data.get('show_name'),
+                             data.get('season'),
+                             exception_id])
+        if cache_db_con.connection.total_changes - last_changes == 1:
+            return self.api_finish(status=204)
+        return self.api_finish(status=400, error="Could not update.")
+
+    def post(self, *args, **kwargs):
+        """Add a show."""
+        data = json_decode(self.request.body)
+
+        if not all([data.get('indexer_id'),
+                    data.get('season'),
+                    data.get('show_name'),
+                    data.get('indexer')]):
+            return self.api_finish(status=400, error="Invalid post body, can't update")
+
+        cache_db_con = db.DBConnection('cache.db')
+        last_changes = cache_db_con.connection.total_changes
+        cache_db_con.action(b'INSERT INTO scene_exceptions'
+                            b' (indexer, indexer_id, show_name, season, custom) '
+                            b' values (?,?,?,?,1)',
+                            [data.get('indexer'),
+                             data.get('indexer_id'),
+                             data.get('show_name'),
+                             data.get('season')])
+        if cache_db_con.connection.total_changes - last_changes > 0:
+            return self.api_finish(status=204)
```

Comment:
Since `POST` is creation, you can return `201 CREATED`

We're still deciding our conventions, but IMO we should use this [guide](http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#useful-post-responses) as a base for our work. 

>>> In case of a POST that resulted in a creation, use a HTTP 201 status code and include a Location header that points to the URL of the new resource.